### PR TITLE
GitHub pull request template

### PR DESCRIPTION
**Pull request by tautschnig**
The template provides guidance and a check-list to submitters to make sure code
reviews can be as efficient as possible.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!---
+Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
+--->
+
+- [ ] Each commit message has a non-empty body, answering the "Why?" question.
```
Comment (smowton) [inline .github/pull_request_template.md]: "...explaining why the change was made"?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- [ ] My contribution is formatting in line with CODING_STANDARD.md.
```
Comment (smowton) [inline .github/pull_request_template.md]: is formatted

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- [ ] Methods or procedures I have added are documented.
```
Comment (smowton) [inline .github/pull_request_template.md]: ...using Doxygen comments at the definition site (not the declaration)

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+- [ ] Regression or unit tests are included, or existing tests cover the modified code.
```
Comment (smowton) [inline .github/pull_request_template.md]: (and I say which ones)

File: .github/pull_request_template.md (hunk 3 above)
Comment (tautschnig) [reply .github/pull_request_template.md]: I have included a reference to CODING_STANDARD.md.

**Review by smowton — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: caf1f5f1d2311e8cf3f65aeec5c53e839c30916e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85118287
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Comment by tautschnig** [comment]
I am going to merge this once CI passes, despite the TG compatibility check failing: I have no idea why a non-code change should be causing this and thus assume it was broken before already.

**Review by chrisr-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 9a56f42fce0f7cadcf83c4989cbc8d1733c51d7a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85120424
